lib/models/proof_v3.ex

```diff
@@ -31,7 +30,21 @@ defmodule Cashu.ProofV3 do
     end
   end
 
- def validate(_), do: {:ok, nil}
+  def validate(_), do: {:ok, nil}
 
- def validate_proof_list(list), do: Validator.validate_list(list, &validate/1)
+  def validate_proof_list(list), do: Validator.validate_list(list, &validate/1)
+end
+
+defimpl Jason.Encoder, for: Cashu.ProofV3 do
```

Comment:
I did this before asking in the cashu discord. I just wanted my tests to pass and the result to be equal to the output in the tests vectors. For that to be the case, the order of the keys has to preserved as python preserves insertion order.

Reply:
man, this is strange. I can almost get the right encoded json save for a small portion of the data when forcing proofs to encode maps. The reason your thing makes sense is that we need to map `c` to `C` when encoding. But the Jason Encoder should be able to handle a map instead of a keyword list in TokenV3, but it doesn't... And deserialization with vanilla Jason Decoder works fine. hmmm to be continued.